my-dsl.md

```diff
@@ -2,37 +2,79 @@
 _What is the name of the language? Link the name to its webpage 
 (if appropriate)._
 
+AWK, http://www.cs.princeton.edu/~bwk/btl.mirror/
 
 # Domain
 _Describe the language's domain in five words._
 
+Text processing and data extraction
 
 # Computational model
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of 
 what happens when a program in your DSL runs._
 
+AWK runs on pattern-action pairs, written as `pattern { action }`. The pattern 
+is typically an expression, and the action is a series of commands. As an AWK 
+script runs, it tries to match each pattern with the data or text it is run 
+with. If there are any matches, AWK will run the action that is paired with the 
+pattern.
 
 # DSL-ness
 _Fowler writes about a spectrum of languages, from general-purpose languages to 
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
 and why?_ 
 
+AWK was originally designed to process text and manipulate data. Thus, it is 
+mostly used in one-liner scripts/programs that do these domain-specific things. 
+However, since AWK is Turing-complete, it can be used like a general-
+purpose language and has been used to create large AWK programs. 
+
+On the spectrum of languages, it sits between general-purpose and purely domain-
+specific languages, leaning towards purely domain-specific. This is because its 
+intended use was for text manipulation, and it is most often used in one-liner 
+scripts inside larger applications or as part of a programmer's workflow. 
+However, as mentioned above, it is Turing-Complete and it is capable of doing 
+much more than purely text processing and data extraction, so it can be used as 
+a general-purpose language. Fowler argues that languages that are Turing-
+complete should not be considered DSLs. However, Fowler also argues that the 
+way a language is used also matters. AWK is mostly used to do text processing, 
+so it is definitely more on the domain-specific side of things.
 
```

Comment:
I see how it can be hard for one to decide whether AWK is more of a general-purpose language or a DSL, because it's Turing-complete and therefore can be used as a general-purpose language. So I would determine this depending on whether a lot of people actually do use AWK for things other than text manipulation and data extraction. It's not always about whether people CAN, but if they actually DO?